contracts/sg-ics721/src/testing/integration_tests.rs

```diff
@@ -0,0 +1,1440 @@
+                                to_binary(&CollectionData {
+                                    owner: Some(OWNER_SOURCE_CHAIN.to_string()),
+                                    contract_info: Default::default(),
+                                    name: "name".to_string(),
+                                    symbol: "symbol".to_string(),
+                                    num_tokens: 1,
+                                })
+                                .unwrap(),
+                            ),
+                        },
+                        tokens: vec![
+                            Token {
+                                id: TokenId::new("1"),
+                                uri: Some("https://moonphase.is/image.svg".to_string()),
+                                data: None,
+                            },
+                            Token {
+                                id: TokenId::new("2"),
+                                uri: Some("https://moonphase.is/image.svg".to_string()),
+                                data: None,
+                            },
+                        ],
+                    },
+                }),
+                &[],
+            )
+            .unwrap();
+        // Check entry added in CLASS_ID_TO_NFT_CONTRACT
+        let nft_contracts = test.query_nft_contracts();
+        assert_eq!(nft_contracts.len(), 1);
+        assert_eq!(nft_contracts[0].0, "bad kids");
+        // Get the address of the instantiated NFT.
+        let nft: Addr = test
+            .app
+            .wrap()
+            .query_wasm_smart(
+                test.ics721.clone(),
+                &QueryMsg::NftContract {
+                    class_id: "bad kids".to_string(),
+                },
+            )
+            .unwrap();
+
+        // check contract info is properly set
+        let contract_info: cw721::ContractInfoResponse = test
+            .app
+            .wrap()
+            .query_wasm_smart(nft.clone(), &Cw721QueryMsg::<Empty>::ContractInfo {})
+            .unwrap();
+        assert_eq!(
+            contract_info,
+            cw721::ContractInfoResponse {
+                name: "bad kids".to_string(),
+                symbol: "bad kids".to_string()
+            }
+        );
+
+        // check collection info is properly set
+        let collection_info: CollectionInfoResponse = test
+            .app
+            .wrap()
+            .query_wasm_smart(nft.clone(), &Sg721QueryMsg::CollectionInfo {})
+            .unwrap();
+        let (_source_hrp, source_data, source_variant) =
+            bech32::decode(OWNER_SOURCE_CHAIN).unwrap();
+        let target_owner = bech32::encode(TARGET_HRP, source_data, source_variant).unwrap();
+
+        assert_eq!(
+            collection_info,
+            CollectionInfoResponse {
+                creator: target_owner, // creator is set to owner as defined by ClassData
```

Comment:
Test owner from juno converted to SG and sg721 collections holds this as creator info.